v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java

```diff
@@ -0,0 +1,174 @@
+
+  @Before
+  public void setup() throws IOException {
+    // Set up resource managers
+    pubsubResourceManager =
+        DefaultPubsubResourceManager.builder(testName, project)
+            .credentialsProvider(CREDENTIALS_PROVIDER)
+            .build();
+
+    Storage storageClient = createStorageClient(CREDENTIALS);
+    gcsClient = GcsArtifactClient.builder(storageClient, ARTIFACT_BUCKET, TEST_ROOT_DIR).build();
+  }
+
+  @After
+  public void teardown() {
+    ResourceManagerUtils.cleanResources(pubsubResourceManager, gcsClient);
+  }
+
+  @Test
+  public void testBacklog10MB() throws IOException, ParseException, InterruptedException {
+    testBacklog(
+        Function.identity(), NUM_MESSAGES_FOR_10_MB_TEST, TIMEOUT_FOR_10_MB_TEST_IN_MINUTES);
+  }
+
+  @Test
+  public void testBacklog10GB() throws IOException, ParseException, InterruptedException {
+    testBacklog(
+        Function.identity(), NUM_MESSAGES_FOR_10_GB_TEST, TIMEOUT_FOR_10_GB_TEST_IN_MINUTES);
+  }
+
+  public void testBacklog(
+      Function<PipelineLauncher.LaunchConfig.Builder, PipelineLauncher.LaunchConfig.Builder>
+          paramsAdder,
+      long numMessages,
+      long timeoutMin)
+      throws IOException, InterruptedException, ParseException {
+    TopicName backlogTopic = pubsubResourceManager.createTopic("backlog-input");
+
+    DataGenerator dataGenerator =
+        DataGenerator.builderWithSchemaTemplate(testName, "GAME_EVENT")
+            .setQPS("1000000")
+            .setMessagesLimit(String.valueOf(numMessages))
+            .setTopic(backlogTopic.toString())
+            .setNumWorkers("50")
+            .setMaxNumWorkers("100")
+            .build();
+
+    PipelineLauncher.LaunchConfig options =
+        paramsAdder
+            .apply(
+                PipelineLauncher.LaunchConfig.builder(testName, SPEC_PATH)
+                    .addParameter("inputTopic", backlogTopic.toString())
+                    .addParameter(
+                        "outputDirectory",
+                        ArtifactUtils.getFullGcsPath(
+                            ARTIFACT_BUCKET,
+                            getClass().getSimpleName(),
+                            gcsClient.runId(),
+                            testName))
+                    .addParameter(
+                        "avroTempDirectory",
+                        ArtifactUtils.getFullGcsPath(
+                            ARTIFACT_BUCKET,
+                            getClass().getSimpleName(),
+                            gcsClient.runId(),
+                            AVRO_TEMP_DIR))
+                    .addParameter("outputFilenamePrefix", AVRO_OUTPUT_FILENAME_PREFIX))
+            .build();
+
+    // Act
+    PipelineLauncher.LaunchInfo info = pipelineLauncher.launch(project, region, options);
+    assertThatPipeline(info).isRunning();
+
+    // Wait for the template to start
+    TimeUnit.SECONDS.sleep(WAIT_FOR_WORKERS_TIME_SEC);
+    dataGenerator.execute(Duration.ofMinutes(timeoutMin));
```

Comment:
Just realized that moving the datagenerator before running the pipeline would result in no messages in the Pub/Sub to Avro pipeline (since the pipeline does not support passing in a subscription due to https://github.com/GoogleCloudPlatform/DataflowTemplates/issues/723, and if a topic is provided it creates a new subscription)

I'm working on solving that issue

Reply:
Let's change the current test and update the spec path to `gs://dataflow-templates-southamerica-west1/latest/flex/Cloud_PubSub_to_Avro_Flex`

Let's also move this test to `v2/googlecloud-to-googlecloud`,

Reply:
Done